client/src/dataset/components/ScanSize.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from "react";
+import { Button, Form } from "semantic-ui-react";
+import { parseNumListWithPadding } from "../helpers";
+import ScanSizePart from "./ScanSizePart";
+
+interface ScanSizeProps {
+    value: string,
+    minScan: number,
+    maxScan: number,
+    setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void,
+}
+
+const ScanSize: React.FC<ScanSizeProps> = ({ value, minScan, maxScan, setFieldValue }) => {
+
+    const [scanSize, setScanSize] = React.useState(parseNumListWithPadding(value, minScan));
+
+    React.useEffect(() => {
+      const newScanSizeValue = parseNumListWithPadding(value, minScan);
+      setScanSize(newScanSizeValue);
+    }, [value, minScan]);
+
+    const scanRefsArray = React.useRef<HTMLInputElement[]>([]);
+
+    const scanSizeChangeHandle = (idx: number, val: string) => {
+      const newScanSize = [...scanSize];
+      newScanSize[idx] = val;
+      setScanSize(newScanSize);
+      setFieldValue("scan_size", newScanSize.toString());
+    };
+
+    const commaPressHandle = (idx: number) => {
+      if(idx===(scanSize.length-1)) {
+        newScanDim();
+      } else {
+          scanRefsArray.current[idx+1].focus();
+      }
+    }
+
+    const newScanDim = () => {
+      if(scanSize.length < maxScan) {
+        const newScanSize = [...scanSize];
+        newScanSize.push("");
+        setScanSize(newScanSize);
+        setFieldValue("scan_size", newScanSize.toString());
+      }
+    }
+
+    React.useEffect(() => {
+      if(scanSize.length > minScan) {
+        scanRefsArray.current[scanSize.length-1].focus();
+      }
+    }, [scanSize.length, minScan]);
+
+    const delScanDim = () => {
+      if(scanSize.length > minScan) {
+        const newScanSize = [...scanSize];
+        newScanSize.pop();
+        setScanSize(newScanSize);
+        setFieldValue("scan_size", newScanSize.toString());
+      }
+    }
+
+    return (
+      <>
+        <Form.Group>
+          {parseNumListWithPadding(value, minScan).map((val, idx) => {
+            const scanRef = (ref:HTMLInputElement) => { scanRefsArray.current[idx] = ref; }
+            return <Form.Field width={2} key={idx}><ScanSizePart scanKey={idx} name={"scan_size_" + idx} id={"id_scan_size_" + idx} value={val} scanRef={scanRef} scanSizeChangeHandle={scanSizeChangeHandle} commaPressHandle={commaPressHandle} /></Form.Field>
+          })}
+          <Form.Field hidden={minScan === maxScan}>
+            <Button onClick={newScanDim} disabled={scanSize.length === maxScan} type="button" icon="add" title="Add dimension" basic={true} color="blue" />
+            <Button onClick={delScanDim} disabled={scanSize.length === minScan} type="button" icon="minus" title="Remove dimension" basic={true} color="blue" />
```

Comment:
A comment about the buttons: keyboard focus is very hard to see on `basic={true}` buttons. You can check this by focusing the last dimension and pressing tab. Currently, the shade of the "+" changes veeery sliiiightly.

Even though it might be a tiny bit more ugly, what do you think about using `basic={false}` buttons here with the default grey color? That one makes the keyboard focus obvious.

Thanks for the great work so far, I think this is almost done!